src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -1129,3 +1267,60 @@ def _get_iot_central_app_by_name(client, app_name):
         raise CLIError(
             "No IoT Central application found with name {} in current subscription.".format(app_name))
     return target_app
+
+
+def _process_fileupload_args(
+        default_storage_endpoint,
+        fileupload_storage_connectionstring=None,
+        fileupload_storage_container_name=None,
+        fileupload_sas_ttl=None,
+        fileupload_storage_authentication_type=None,
+        fileupload_storage_container_uri=None,
+        fileupload_storage_identity=None,
+):
+    from datetime import timedelta
+    if fileupload_storage_authentication_type and fileupload_storage_authentication_type.lower() == AuthenticationType.IdentityBased.value:
+        default_storage_endpoint.authentication_type = AuthenticationType.IdentityBased
+        default_storage_endpoint.container_uri = fileupload_storage_container_uri
+    elif fileupload_storage_authentication_type is not None:
+        default_storage_endpoint.authentication_type = None
+        default_storage_endpoint.container_uri = None
+    # TODO - remove connection string and set containerURI once fileUpload SAS URL is enabled
+    if fileupload_storage_connectionstring is not None and fileupload_storage_container_name is not None:
+        default_storage_endpoint.connection_string = fileupload_storage_connectionstring
+        default_storage_endpoint.container_name = fileupload_storage_container_name
+    elif fileupload_storage_connectionstring is not None:
+        raise CLIError('Please mention storage container name.')
+    elif fileupload_storage_container_name is not None:
+        raise CLIError('Please mention storage connection string.')
+    if fileupload_sas_ttl is not None:
+        default_storage_endpoint.sas_ttl_as_iso8601 = timedelta(hours=fileupload_sas_ttl)
+
+    # Fix for identity/authentication-type params missing on hybrid profile api
+    if hasattr(default_storage_endpoint, 'authentication_type'):
+        # If we are now (or will be) using fsa=identity AND we've set a new identity
+        if default_storage_endpoint.authentication_type == AuthenticationType.IdentityBased and fileupload_storage_identity:
+            # setup new fsi
+            default_storage_endpoint.identity = ManagedIdentity(user_assigned_identity=fileupload_storage_identity) if fileupload_storage_identity not in [IdentityType.none.value, SYSTEM_IDENTITY] else None
+        # otherwise - let them know they need identity-based auth enabled
+        elif fileupload_storage_identity:
+            raise CLIError('In order to set a file upload storage identity, you must set the file upload storage authentication type (--fsa) to IdentityBased')
```

Comment:
Please use the specific error type `RequiredArgumentMissingError` instead of `CLIError`.